docs/go_api.md

```diff
@@ -0,0 +1,342 @@
+# WasmEdge Go API Documentation
+
+The followings are the guides to working with the WasmEdge-Go.
+
+## Table of Contents
+
+* [Getting Started](#Getting-Started)
+* [WasmEdge-go Extensions](#WasmEdge-go-Extensions)
+* [Go-API Examples](#Go-API-Examples)
+	* [Embed a function](#Embed-a-function)
+	* [Embed a full program](#Embed-a-full-program)
+	* [WasmEdge AOT Compiler in GO](#WasmEdge-AOT-Compiler-in-GO)
+
+## Getting Started
+
+The WasmEdge-go requires golang version >= 1.15. Please check your golang version before installation. Developers can [download golang here](https://golang.org/dl/).
+
+```bash
+$ go version
+go version go1.16.5 linux/amd64
+```
+
+Developers must [install the WasmEdge shared library](https://github.com/WasmEdge/WasmEdge/blob/master/docs/install.md) with the same `WasmEdge-go` release version.
+
+```bash
+wget -qO- https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -v 0.9.0-rc.2
```

Comment:
Which version would you like to use in this documentation? `0.9.0-rc.2` or `0.8.2`?

Reply:
For this document, I think use the `0.9.0-rc.2` and `WasmEdge-go v0.9.0-rc3` is better.

Reply:
I have updated the document. Please have a look.